api_app/api/routes/airlock.py

```diff
@@ -35,3 +37,17 @@ async def create_draft_request(airlock_request_input: AirlockRequestInCreate, us
 async def create_submit_request(airlock_request=Depends(get_airlock_request_by_id_from_path), user=Depends(get_current_workspace_owner_or_researcher_user), airlock_request_repo=Depends(get_repository(AirlockRequestRepository))) -> AirlockRequestInResponse:
     updated_resource = await update_status_and_publish_event_airlock_request(airlock_request, airlock_request_repo, user, AirlockRequestStatus.Submitted)
     return AirlockRequestInResponse(airlock_request=updated_resource)
+
+
+@airlock_workspace_router.post("/workspaces/{workspace_id}/requests/{airlock_request_id}/reviews", status_code=status.HTTP_200_OK, response_model=AirlockReviewInResponse, name=strings.API_REVIEW_AIRLOCK_REQUEST, dependencies=[Depends(get_workspace_by_id_from_path)])
+async def create_airlock_review(airlock_review_input: AirlockReviewInCreate, airlock_request=Depends(get_airlock_request_by_id_from_path), user=Depends(get_current_workspace_owner_or_researcher_user), airlock_request_repo=Depends(get_repository(AirlockRequestRepository)), airlock_review_repo=Depends(get_repository(AirlockReviewRepository)), workspace=Depends(get_deployed_workspace_by_id_from_path)) -> AirlockReviewInResponse:
+    # Create the review model and save in cosmos
+    try:
+        airlock_review = airlock_review_repo.create_airlock_review_item(airlock_review_input, workspace.id, airlock_request.id)
+    except (ValidationError, ValueError) as e:
+        logging.error(f"Failed create airlock review model instance: {e}")
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
+    # Update the airlock request in cosmos, send a status_changed event
+    await update_status_and_publish_event_airlock_request(airlock_request, airlock_request_repo, user, airlock_review.reviewDecision)
+    await save_airlock_review(airlock_review, airlock_review_repo, user)
```

Comment:
dont you want to first save the review.
imagine the saving of the review object fails, so in the current code, you already continued the flow, and the review is not saved == cant trace back.

prefer to save the review first, worst case the flow fails later, but it didnt continue until the trigger was saved

wdyt